### PR TITLE
Add Initiate, Get and Accept Subscription Change Directory Api with new version.

### PR DESCRIPTION
**Pull request by abhaygill338**
# Add Initiate, Get and Accept Subscription Change Directory Api with new version

Switch to "Preview" on this description then select one of the choices below.

<a href="?expand=1&template=data_plane_template.md">Click here</a> to open a PR for a Data Plane API.

<a href="?expand=1&template=control_plane_template.md">Click here</a> to open a PR for a Control Plane (ARM) API.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,5 +41,13 @@
     "prettier": "prettier",
     "tsc": "tsc",
     "multiapi": "ts-node ./scripts/multiapi.ts"
+  },
```
Comment (weshaggard) [inline package.json:44]: Please remove the changes to this file and the package-lock.json

Comment (abhaygill338) [reply package.json]: Hi @weshaggard , remove the changes from the package file and package-lock.json file.

**Comment by ms-bogdan-rotaru** [comment]
The first commit needs to be an exact copy of the previous api version. All new changes should only be added in the subsequent commits. With the way the PR is raised now, it is not possible for me to tell what the changes are. Please either abandon the PR and raise another one with the recommendation or create a new set of commits on this PR following the recommendation.

**Comment by gary-x-li** [comment]
@abhaygill338 Can you please respond to this comment? https://github.com/Azure/azure-rest-api-specs/pull/29912#issuecomment-2255647910

**Comment by abhaygill338** [comment]
> The first commit needs to be an exact copy of the previous api version. All new changes should only be added in the subsequent commits. With the way the PR is raised now, it is not possible for me to tell what the changes are. Please either abandon the PR and raise another one with the recommendation or create a new set of commits on this PR following the recommendation.

**Comment by gary-x-li** [comment]
Note to reviewers: The base version of `subscriptions.json` is [2021-10-01](https://github.com/Azure/azure-rest-api-specs/blob/5f81e1bdb315630a90262e752946b534236e32c8/specification/subscription/resource-manager/Microsoft.Subscription/stable/2021-10-01/subscriptions.json) according to the author.

File: specification/subscription/resource-manager/readme.md [hunk 2]
```diff
@@ -23,6 +23,13 @@ To see additional help and options, run:
 ``` yaml
 directive:
   - suppress: R2059
+  - suppress: PutResponseCodes
+    from: subscriptions.json
+    where: $.paths["/subscriptions/{subscriptionId}/providers/Microsoft.Subscription/changeTenantRequest/default"].put
+    reason: Not supposed to return 201 as the response code for the below API since existing api with new version change, got exceptions from ARM reviewer.
```
Comment (gary-x-li) [inline specification/subscription/resource-manager/readme.md:29]: I'm not very clear on the reason here. Is this API introduced in this PR? I didn't find this API in the `2019-10-01-preview` version.

File: specification/subscription/resource-manager/readme.md [hunk 3]
```diff
@@ -23,6 +23,13 @@ To see additional help and options, run:
...
+  - suppress: PutRequestResponseSchemeArm
```
Comment (gary-x-li) [inline specification/subscription/resource-manager/readme.md:30]: Have you got an exception from ARM reviewer as well?

File: specification/subscription/resource-manager/readme.md (hunk 2 above)
Comment (abhaygill338) [reply specification/subscription/resource-manager/readme.md:29]: @gary-x-li , it's actually not present in our public swagger, but we are using the api in 2019-10-01-preview , example - https://msazure.visualstudio.com/One/_git/AD-CAB-Provisioning?path=/src/WebApps/SubscriptionRP/Controllers/V2019_10_01/TargetDirectoryController.cs&version=GBdev&line=58&lineEnd=61&lineStartColumn=9&lineEndColumn=67&lineStyle=plain&_a=contents, 
Didn't know why the swagger is not present in an old preview version, probably missed adding it or there are some different process that time.

File: specification/subscription/resource-manager/Microsoft.Subscription/preview/2024-08-01-preview/subscriptions.json [hunk 4]
```diff
@@ -0,0 +1,1396 @@
...
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/TargetDirectoryResult"
+            }
+          },
+          "default": {
+            "description": "Describes the error if the operation is not successful.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponseBody"
```
Comment (gary-x-li) [inline specification/subscription/resource-manager/Microsoft.Subscription/preview/2024-08-01-preview/subscriptions.json:593]: Use `ErrorResponse` from common types? I saw you made the change in some places. Can you change them all?

Comment (abhaygill338) [reply specification/subscription/resource-manager/Microsoft.Subscription/preview/2024-08-01-preview/subscriptions.json:593]: sure @gary-x-li

File: specification/subscription/resource-manager/readme.md (hunk 3 above)
Comment (abhaygill338) [reply specification/subscription/resource-manager/readme.md:30]: Yes @gary-x-li , got an Exception from Roopesh to suppress with the reason provided.

**Review by gary-x-li — APPROVED** [review]

**Pull request merged**